components/script/dom/oscillatornode.rs

```diff
@@ -113,6 +114,24 @@ impl OscillatorNodeMethods for OscillatorNode {
     fn Detune(&self) -> DomRoot<AudioParam> {
         DomRoot::from_ref(&self.detune)
     }
+
+    // https://webaudio.github.io/web-audio-api/#dom-oscillatornode-type
+    fn Type(&self) -> OscillatorType {
+        self.oscillator_type.get()
+    }
+
+    // https://webaudio.github.io/web-audio-api/#dom-oscillatornode-type
+    fn SetType(&self, type_: OscillatorType) -> ErrorResult {
+        self.oscillator_type.set(type_);
+        if type_ == "custom" {
+            return Err(Error::InvalidState);
+        }
+        self.source_node
+            .node()
+            .message(AudioNodeMessage::OscillatorNode(
+                OscillatorNodeMessage::SetOscillatorType(type_.into()),
+            ));
+    }
```

Comment:
```
error[E0308]: mismatched types
   --> components/script/dom/oscillatornode.rs:124:49
    |
124 |     fn SetType(&self, type_: OscillatorType) -> ErrorResult {
    |        -------                                  ^^^^^^^^^^^ expected enum `std::result::Result`, found ()
    |        |
    |        this function's body doesn't return
    |
    = note: expected type `std::result::Result<(), dom::bindings::error::Error>`
               found type `()`
```

This function expects an `ErrorResult` and at this point everthing should be fine, so we need to return `Ok(())` here.